// @aws-sdk/client-rekognition/dist-cjs/index.js
'use strict';

var client$1 = require('@aws-sdk/core/client');
var core = require('@smithy/core');
var client = require('@smithy/core/client');
var config = require('@smithy/core/config');
var endpoints = require('@smithy/core/endpoints');
var protocols = require('@smithy/core/protocols');
var retry = require('@smithy/core/retry');
var schema = require('@smithy/core/schema');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var schemas_0 = require('./schemas/schemas_0');
var errors = require('./models/errors');
var RekognitionServiceException = require('./models/RekognitionServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "rekognition",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(client$1.getAwsRegionExtensionConfiguration(runtimeConfig), client.getDefaultExtensionConfiguration(runtimeConfig), protocols.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, client$1.resolveAwsRegionExtensionConfiguration(extensionConfiguration), client.resolveDefaultRuntimeConfig(extensionConfiguration), protocols.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class RekognitionClient extends client.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = client$1.resolveUserAgentConfig(_config_1);
        const _config_3 = retry.resolveRetryConfig(_config_2);
        const _config_4 = config.resolveRegionConfig(_config_3);
        const _config_5 = client$1.resolveHostHeaderConfig(_config_4);
        const _config_6 = endpoints.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(client$1.getUserAgentPlugin(this.config));
        this.middlewareStack.use(retry.getRetryPlugin(this.config));
        this.middlewareStack.use(protocols.getContentLengthPlugin(this.config));
        this.middlewareStack.use(client$1.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(client$1.getLoggerPlugin(this.config));
        this.middlewareStack.use(client$1.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultRekognitionHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class AssociateFacesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "AssociateFaces", {})
    .n("RekognitionClient", "AssociateFacesCommand")
    .sc(schemas_0.AssociateFaces$)
    .build() {
}

class CompareFacesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "CompareFaces", {})
    .n("RekognitionClient", "CompareFacesCommand")
    .sc(schemas_0.CompareFaces$)
    .build() {
}

class CopyProjectVersionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "CopyProjectVersion", {})
    .n("RekognitionClient", "CopyProjectVersionCommand")
    .sc(schemas_0.CopyProjectVersion$)
    .build() {
}

class CreateCollectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "CreateCollection", {})
    .n("RekognitionClient", "CreateCollectionCommand")
    .sc(schemas_0.CreateCollection$)
    .build() {
}

class CreateDatasetCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "CreateDataset", {})
    .n("RekognitionClient", "CreateDatasetCommand")
    .sc(schemas_0.CreateDataset$)
    .build() {
}

class CreateFaceLivenessSessionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "CreateFaceLivenessSession", {})
    .n("RekognitionClient", "CreateFaceLivenessSessionCommand")
    .sc(schemas_0.CreateFaceLivenessSession$)
    .build() {
}

class CreateProjectCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "CreateProject", {})
    .n("RekognitionClient", "CreateProjectCommand")
    .sc(schemas_0.CreateProject$)
    .build() {
}

class CreateProjectVersionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "CreateProjectVersion", {})
    .n("RekognitionClient", "CreateProjectVersionCommand")
    .sc(schemas_0.CreateProjectVersion$)
    .build() {
}

class CreateStreamProcessorCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "CreateStreamProcessor", {})
    .n("RekognitionClient", "CreateStreamProcessorCommand")
    .sc(schemas_0.CreateStreamProcessor$)
    .build() {
}

class CreateUserCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "CreateUser", {})
    .n("RekognitionClient", "CreateUserCommand")
    .sc(schemas_0.CreateUser$)
    .build() {
}

class DeleteCollectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DeleteCollection", {})
    .n("RekognitionClient", "DeleteCollectionCommand")
    .sc(schemas_0.DeleteCollection$)
    .build() {
}

class DeleteDatasetCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DeleteDataset", {})
    .n("RekognitionClient", "DeleteDatasetCommand")
    .sc(schemas_0.DeleteDataset$)
    .build() {
}

class DeleteFacesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DeleteFaces", {})
    .n("RekognitionClient", "DeleteFacesCommand")
    .sc(schemas_0.DeleteFaces$)
    .build() {
}

class DeleteProjectCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DeleteProject", {})
    .n("RekognitionClient", "DeleteProjectCommand")
    .sc(schemas_0.DeleteProject$)
    .build() {
}

class DeleteProjectPolicyCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DeleteProjectPolicy", {})
    .n("RekognitionClient", "DeleteProjectPolicyCommand")
    .sc(schemas_0.DeleteProjectPolicy$)
    .build() {
}

class DeleteProjectVersionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DeleteProjectVersion", {})
    .n("RekognitionClient", "DeleteProjectVersionCommand")
    .sc(schemas_0.DeleteProjectVersion$)
    .build() {
}

class DeleteStreamProcessorCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DeleteStreamProcessor", {})
    .n("RekognitionClient", "DeleteStreamProcessorCommand")
    .sc(schemas_0.DeleteStreamProcessor$)
    .build() {
}

class DeleteUserCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DeleteUser", {})
    .n("RekognitionClient", "DeleteUserCommand")
    .sc(schemas_0.DeleteUser$)
    .build() {
}

class DescribeCollectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DescribeCollection", {})
    .n("RekognitionClient", "DescribeCollectionCommand")
    .sc(schemas_0.DescribeCollection$)
    .build() {
}

class DescribeDatasetCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DescribeDataset", {})
    .n("RekognitionClient", "DescribeDatasetCommand")
    .sc(schemas_0.DescribeDataset$)
    .build() {
}

class DescribeProjectsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DescribeProjects", {})
    .n("RekognitionClient", "DescribeProjectsCommand")
    .sc(schemas_0.DescribeProjects$)
    .build() {
}

class DescribeProjectVersionsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DescribeProjectVersions", {})
    .n("RekognitionClient", "DescribeProjectVersionsCommand")
    .sc(schemas_0.DescribeProjectVersions$)
    .build() {
}

class DescribeStreamProcessorCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DescribeStreamProcessor", {})
    .n("RekognitionClient", "DescribeStreamProcessorCommand")
    .sc(schemas_0.DescribeStreamProcessor$)
    .build() {
}

class DetectCustomLabelsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DetectCustomLabels", {})
    .n("RekognitionClient", "DetectCustomLabelsCommand")
    .sc(schemas_0.DetectCustomLabels$)
    .build() {
}

class DetectFacesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DetectFaces", {})
    .n("RekognitionClient", "DetectFacesCommand")
    .sc(schemas_0.DetectFaces$)
    .build() {
}

class DetectLabelsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DetectLabels", {})
    .n("RekognitionClient", "DetectLabelsCommand")
    .sc(schemas_0.DetectLabels$)
    .build() {
}

class DetectModerationLabelsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DetectModerationLabels", {})
    .n("RekognitionClient", "DetectModerationLabelsCommand")
    .sc(schemas_0.DetectModerationLabels$)
    .build() {
}

class DetectProtectiveEquipmentCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DetectProtectiveEquipment", {})
    .n("RekognitionClient", "DetectProtectiveEquipmentCommand")
    .sc(schemas_0.DetectProtectiveEquipment$)
    .build() {
}

class DetectTextCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DetectText", {})
    .n("RekognitionClient", "DetectTextCommand")
    .sc(schemas_0.DetectText$)
    .build() {
}

class DisassociateFacesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DisassociateFaces", {})
    .n("RekognitionClient", "DisassociateFacesCommand")
    .sc(schemas_0.DisassociateFaces$)
    .build() {
}

class DistributeDatasetEntriesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "DistributeDatasetEntries", {})
    .n("RekognitionClient", "DistributeDatasetEntriesCommand")
    .sc(schemas_0.DistributeDatasetEntries$)
    .build() {
}

class GetCelebrityInfoCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "GetCelebrityInfo", {})
    .n("RekognitionClient", "GetCelebrityInfoCommand")
    .sc(schemas_0.GetCelebrityInfo$)
    .build() {
}

class GetCelebrityRecognitionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "GetCelebrityRecognition", {})
    .n("RekognitionClient", "GetCelebrityRecognitionCommand")
    .sc(schemas_0.GetCelebrityRecognition$)
    .build() {
}

class GetContentModerationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "GetContentModeration", {})
    .n("RekognitionClient", "GetContentModerationCommand")
    .sc(schemas_0.GetContentModeration$)
    .build() {
}

class GetFaceDetectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "GetFaceDetection", {})
    .n("RekognitionClient", "GetFaceDetectionCommand")
    .sc(schemas_0.GetFaceDetection$)
    .build() {
}

class GetFaceLivenessSessionResultsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "GetFaceLivenessSessionResults", {})
    .n("RekognitionClient", "GetFaceLivenessSessionResultsCommand")
    .sc(schemas_0.GetFaceLivenessSessionResults$)
    .build() {
}

class GetFaceSearchCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "GetFaceSearch", {})
    .n("RekognitionClient", "GetFaceSearchCommand")
    .sc(schemas_0.GetFaceSearch$)
    .build() {
}

class GetLabelDetectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "GetLabelDetection", {})
    .n("RekognitionClient", "GetLabelDetectionCommand")
    .sc(schemas_0.GetLabelDetection$)
    .build() {
}

class GetMediaAnalysisJobCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "GetMediaAnalysisJob", {})
    .n("RekognitionClient", "GetMediaAnalysisJobCommand")
    .sc(schemas_0.GetMediaAnalysisJob$)
    .build() {
}

class GetPersonTrackingCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "GetPersonTracking", {})
    .n("RekognitionClient", "GetPersonTrackingCommand")
    .sc(schemas_0.GetPersonTracking$)
    .build() {
}

class GetSegmentDetectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "GetSegmentDetection", {})
    .n("RekognitionClient", "GetSegmentDetectionCommand")
    .sc(schemas_0.GetSegmentDetection$)
    .build() {
}

class GetTextDetectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "GetTextDetection", {})
    .n("RekognitionClient", "GetTextDetectionCommand")
    .sc(schemas_0.GetTextDetection$)
    .build() {
}

class IndexFacesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "IndexFaces", {})
    .n("RekognitionClient", "IndexFacesCommand")
    .sc(schemas_0.IndexFaces$)
    .build() {
}

class ListCollectionsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "ListCollections", {})
    .n("RekognitionClient", "ListCollectionsCommand")
    .sc(schemas_0.ListCollections$)
    .build() {
}

class ListDatasetEntriesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "ListDatasetEntries", {})
    .n("RekognitionClient", "ListDatasetEntriesCommand")
    .sc(schemas_0.ListDatasetEntries$)
    .build() {
}

class ListDatasetLabelsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "ListDatasetLabels", {})
    .n("RekognitionClient", "ListDatasetLabelsCommand")
    .sc(schemas_0.ListDatasetLabels$)
    .build() {
}

class ListFacesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "ListFaces", {})
    .n("RekognitionClient", "ListFacesCommand")
    .sc(schemas_0.ListFaces$)
    .build() {
}

class ListMediaAnalysisJobsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "ListMediaAnalysisJobs", {})
    .n("RekognitionClient", "ListMediaAnalysisJobsCommand")
    .sc(schemas_0.ListMediaAnalysisJobs$)
    .build() {
}

class ListProjectPoliciesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "ListProjectPolicies", {})
    .n("RekognitionClient", "ListProjectPoliciesCommand")
    .sc(schemas_0.ListProjectPolicies$)
    .build() {
}

class ListStreamProcessorsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "ListStreamProcessors", {})
    .n("RekognitionClient", "ListStreamProcessorsCommand")
    .sc(schemas_0.ListStreamProcessors$)
    .build() {
}

class ListTagsForResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "ListTagsForResource", {})
    .n("RekognitionClient", "ListTagsForResourceCommand")
    .sc(schemas_0.ListTagsForResource$)
    .build() {
}

class ListUsersCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "ListUsers", {})
    .n("RekognitionClient", "ListUsersCommand")
    .sc(schemas_0.ListUsers$)
    .build() {
}

class PutProjectPolicyCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "PutProjectPolicy", {})
    .n("RekognitionClient", "PutProjectPolicyCommand")
    .sc(schemas_0.PutProjectPolicy$)
    .build() {
}

class RecognizeCelebritiesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "RecognizeCelebrities", {})
    .n("RekognitionClient", "RecognizeCelebritiesCommand")
    .sc(schemas_0.RecognizeCelebrities$)
    .build() {
}

class SearchFacesByImageCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "SearchFacesByImage", {})
    .n("RekognitionClient", "SearchFacesByImageCommand")
    .sc(schemas_0.SearchFacesByImage$)
    .build() {
}

class SearchFacesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "SearchFaces", {})
    .n("RekognitionClient", "SearchFacesCommand")
    .sc(schemas_0.SearchFaces$)
    .build() {
}

class SearchUsersByImageCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "SearchUsersByImage", {})
    .n("RekognitionClient", "SearchUsersByImageCommand")
    .sc(schemas_0.SearchUsersByImage$)
    .build() {
}

class SearchUsersCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "SearchUsers", {})
    .n("RekognitionClient", "SearchUsersCommand")
    .sc(schemas_0.SearchUsers$)
    .build() {
}

class StartCelebrityRecognitionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StartCelebrityRecognition", {})
    .n("RekognitionClient", "StartCelebrityRecognitionCommand")
    .sc(schemas_0.StartCelebrityRecognition$)
    .build() {
}

class StartContentModerationCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StartContentModeration", {})
    .n("RekognitionClient", "StartContentModerationCommand")
    .sc(schemas_0.StartContentModeration$)
    .build() {
}

class StartFaceDetectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StartFaceDetection", {})
    .n("RekognitionClient", "StartFaceDetectionCommand")
    .sc(schemas_0.StartFaceDetection$)
    .build() {
}

class StartFaceSearchCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StartFaceSearch", {})
    .n("RekognitionClient", "StartFaceSearchCommand")
    .sc(schemas_0.StartFaceSearch$)
    .build() {
}

class StartLabelDetectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StartLabelDetection", {})
    .n("RekognitionClient", "StartLabelDetectionCommand")
    .sc(schemas_0.StartLabelDetection$)
    .build() {
}

class StartMediaAnalysisJobCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StartMediaAnalysisJob", {})
    .n("RekognitionClient", "StartMediaAnalysisJobCommand")
    .sc(schemas_0.StartMediaAnalysisJob$)
    .build() {
}

class StartPersonTrackingCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StartPersonTracking", {})
    .n("RekognitionClient", "StartPersonTrackingCommand")
    .sc(schemas_0.StartPersonTracking$)
    .build() {
}

class StartProjectVersionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StartProjectVersion", {})
    .n("RekognitionClient", "StartProjectVersionCommand")
    .sc(schemas_0.StartProjectVersion$)
    .build() {
}

class StartSegmentDetectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StartSegmentDetection", {})
    .n("RekognitionClient", "StartSegmentDetectionCommand")
    .sc(schemas_0.StartSegmentDetection$)
    .build() {
}

class StartStreamProcessorCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StartStreamProcessor", {})
    .n("RekognitionClient", "StartStreamProcessorCommand")
    .sc(schemas_0.StartStreamProcessor$)
    .build() {
}

class StartTextDetectionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StartTextDetection", {})
    .n("RekognitionClient", "StartTextDetectionCommand")
    .sc(schemas_0.StartTextDetection$)
    .build() {
}

class StopProjectVersionCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StopProjectVersion", {})
    .n("RekognitionClient", "StopProjectVersionCommand")
    .sc(schemas_0.StopProjectVersion$)
    .build() {
}

class StopStreamProcessorCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "StopStreamProcessor", {})
    .n("RekognitionClient", "StopStreamProcessorCommand")
    .sc(schemas_0.StopStreamProcessor$)
    .build() {
}

class TagResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "TagResource", {})
    .n("RekognitionClient", "TagResourceCommand")
    .sc(schemas_0.TagResource$)
    .build() {
}

class UntagResourceCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "UntagResource", {})
    .n("RekognitionClient", "UntagResourceCommand")
    .sc(schemas_0.UntagResource$)
    .build() {
}

class UpdateDatasetEntriesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "UpdateDatasetEntries", {})
    .n("RekognitionClient", "UpdateDatasetEntriesCommand")
    .sc(schemas_0.UpdateDatasetEntries$)
    .build() {
}

class UpdateStreamProcessorCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("RekognitionService", "UpdateStreamProcessor", {})
    .n("RekognitionClient", "UpdateStreamProcessorCommand")
    .sc(schemas_0.UpdateStreamProcessor$)
    .build() {
}

const paginateDescribeProjects = core.createPaginator(RekognitionClient, DescribeProjectsCommand, "NextToken", "NextToken", "MaxResults");

const paginateDescribeProjectVersions = core.createPaginator(RekognitionClient, DescribeProjectVersionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateGetCelebrityRecognition = core.createPaginator(RekognitionClient, GetCelebrityRecognitionCommand, "NextToken", "NextToken", "MaxResults");

const paginateGetContentModeration = core.createPaginator(RekognitionClient, GetContentModerationCommand, "NextToken", "NextToken", "MaxResults");

const paginateGetFaceDetection = core.createPaginator(RekognitionClient, GetFaceDetectionCommand, "NextToken", "NextToken", "MaxResults");

const paginateGetFaceSearch = core.createPaginator(RekognitionClient, GetFaceSearchCommand, "NextToken", "NextToken", "MaxResults");

const paginateGetLabelDetection = core.createPaginator(RekognitionClient, GetLabelDetectionCommand, "NextToken", "NextToken", "MaxResults");

const paginateGetPersonTracking = core.createPaginator(RekognitionClient, GetPersonTrackingCommand, "NextToken", "NextToken", "MaxResults");

const paginateGetSegmentDetection = core.createPaginator(RekognitionClient, GetSegmentDetectionCommand, "NextToken", "NextToken", "MaxResults");

const paginateGetTextDetection = core.createPaginator(RekognitionClient, GetTextDetectionCommand, "NextToken", "NextToken", "MaxResults");

const paginateListCollections = core.createPaginator(RekognitionClient, ListCollectionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListDatasetEntries = core.createPaginator(RekognitionClient, ListDatasetEntriesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListDatasetLabels = core.createPaginator(RekognitionClient, ListDatasetLabelsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListFaces = core.createPaginator(RekognitionClient, ListFacesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListMediaAnalysisJobs = core.createPaginator(RekognitionClient, ListMediaAnalysisJobsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListProjectPolicies = core.createPaginator(RekognitionClient, ListProjectPoliciesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListStreamProcessors = core.createPaginator(RekognitionClient, ListStreamProcessorsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListUsers = core.createPaginator(RekognitionClient, ListUsersCommand, "NextToken", "NextToken", "MaxResults");

const checkState$1 = async (client$1, input) => {
    let reason;
    try {
        let result = await client$1.send(new DescribeProjectVersionsCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.ProjectVersionDescriptions);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.Status;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "RUNNING");
            }
            if (allStringEq_5) {
                return { state: client.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.ProjectVersionDescriptions);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.Status;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "FAILED") {
                    return { state: client.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: client.WaiterState.RETRY, reason };
};
const waitForProjectVersionRunning = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    return client.createWaiter({ ...serviceDefaults, ...params }, input, checkState$1);
};
const waitUntilProjectVersionRunning = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    const result = await client.createWaiter({ ...serviceDefaults, ...params }, input, checkState$1);
    return client.checkExceptions(result);
};

const checkState = async (client$1, input) => {
    let reason;
    try {
        let result = await client$1.send(new DescribeProjectVersionsCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.ProjectVersionDescriptions);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.Status;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "TRAINING_COMPLETED");
            }
            if (allStringEq_5) {
                return { state: client.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.ProjectVersionDescriptions);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.Status;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "TRAINING_FAILED") {
                    return { state: client.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: client.WaiterState.RETRY, reason };
};
const waitForProjectVersionTrainingCompleted = async (params, input) => {
    const serviceDefaults = { minDelay: 120, maxDelay: 120 };
    return client.createWaiter({ ...serviceDefaults, ...params }, input, checkState);
};
const waitUntilProjectVersionTrainingCompleted = async (params, input) => {
    const serviceDefaults = { minDelay: 120, maxDelay: 120 };
    const result = await client.createWaiter({ ...serviceDefaults, ...params }, input, checkState);
    return client.checkExceptions(result);
};

const commands = {
    AssociateFacesCommand,
    CompareFacesCommand,
    CopyProjectVersionCommand,
    CreateCollectionCommand,
    CreateDatasetCommand,
    CreateFaceLivenessSessionCommand,
    CreateProjectCommand,
    CreateProjectVersionCommand,
    CreateStreamProcessorCommand,
    CreateUserCommand,
    DeleteCollectionCommand,
    DeleteDatasetCommand,
    DeleteFacesCommand,
    DeleteProjectCommand,
    DeleteProjectPolicyCommand,
    DeleteProjectVersionCommand,
    DeleteStreamProcessorCommand,
    DeleteUserCommand,
    DescribeCollectionCommand,
    DescribeDatasetCommand,
    DescribeProjectsCommand,
    DescribeProjectVersionsCommand,
    DescribeStreamProcessorCommand,
    DetectCustomLabelsCommand,
    DetectFacesCommand,
    DetectLabelsCommand,
    DetectModerationLabelsCommand,
    DetectProtectiveEquipmentCommand,
    DetectTextCommand,
    DisassociateFacesCommand,
    DistributeDatasetEntriesCommand,
    GetCelebrityInfoCommand,
    GetCelebrityRecognitionCommand,
    GetContentModerationCommand,
    GetFaceDetectionCommand,
    GetFaceLivenessSessionResultsCommand,
    GetFaceSearchCommand,
    GetLabelDetectionCommand,
    GetMediaAnalysisJobCommand,
    GetPersonTrackingCommand,
    GetSegmentDetectionCommand,
    GetTextDetectionCommand,
    IndexFacesCommand,
    ListCollectionsCommand,
    ListDatasetEntriesCommand,
    ListDatasetLabelsCommand,
    ListFacesCommand,
    ListMediaAnalysisJobsCommand,
    ListProjectPoliciesCommand,
    ListStreamProcessorsCommand,
    ListTagsForResourceCommand,
    ListUsersCommand,
    PutProjectPolicyCommand,
    RecognizeCelebritiesCommand,
    SearchFacesCommand,
    SearchFacesByImageCommand,
    SearchUsersCommand,
    SearchUsersByImageCommand,
    StartCelebrityRecognitionCommand,
    StartContentModerationCommand,
    StartFaceDetectionCommand,
    StartFaceSearchCommand,
    StartLabelDetectionCommand,
    StartMediaAnalysisJobCommand,
    StartPersonTrackingCommand,
    StartProjectVersionCommand,
    StartSegmentDetectionCommand,
    StartStreamProcessorCommand,
    StartTextDetectionCommand,
    StopProjectVersionCommand,
    StopStreamProcessorCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateDatasetEntriesCommand,
    UpdateStreamProcessorCommand,
};
const paginators = {
    paginateDescribeProjects,
    paginateDescribeProjectVersions,
    paginateGetCelebrityRecognition,
    paginateGetContentModeration,
    paginateGetFaceDetection,
    paginateGetFaceSearch,
    paginateGetLabelDetection,
    paginateGetPersonTracking,
    paginateGetSegmentDetection,
    paginateGetTextDetection,
    paginateListCollections,
    paginateListDatasetEntries,
    paginateListDatasetLabels,
    paginateListFaces,
    paginateListMediaAnalysisJobs,
    paginateListProjectPolicies,
    paginateListStreamProcessors,
    paginateListUsers,
};
const waiters = {
    waitUntilProjectVersionRunning,
    waitUntilProjectVersionTrainingCompleted,
};
class Rekognition extends RekognitionClient {
}
client.createAggregatedClient(commands, Rekognition, { paginators, waiters });

const UnsuccessfulFaceAssociationReason = {
    ASSOCIATED_TO_A_DIFFERENT_USER: "ASSOCIATED_TO_A_DIFFERENT_USER",
    FACE_NOT_FOUND: "FACE_NOT_FOUND",
    LOW_MATCH_CONFIDENCE: "LOW_MATCH_CONFIDENCE",
};
const UserStatus = {
    ACTIVE: "ACTIVE",
    CREATED: "CREATED",
    CREATING: "CREATING",
    UPDATING: "UPDATING",
};
const Attribute = {
    AGE_RANGE: "AGE_RANGE",
    ALL: "ALL",
    BEARD: "BEARD",
    DEFAULT: "DEFAULT",
    EMOTIONS: "EMOTIONS",
    EYEGLASSES: "EYEGLASSES",
    EYES_OPEN: "EYES_OPEN",
    EYE_DIRECTION: "EYE_DIRECTION",
    FACE_OCCLUDED: "FACE_OCCLUDED",
    GENDER: "GENDER",
    MOUTH_OPEN: "MOUTH_OPEN",
    MUSTACHE: "MUSTACHE",
    SMILE: "SMILE",
    SUNGLASSES: "SUNGLASSES",
};
const BodyPart = {
    FACE: "FACE",
    HEAD: "HEAD",
    LEFT_HAND: "LEFT_HAND",
    RIGHT_HAND: "RIGHT_HAND",
};
const ProtectiveEquipmentType = {
    FACE_COVER: "FACE_COVER",
    HAND_COVER: "HAND_COVER",
    HEAD_COVER: "HEAD_COVER",
};
const EmotionName = {
    ANGRY: "ANGRY",
    CALM: "CALM",
    CONFUSED: "CONFUSED",
    DISGUSTED: "DISGUSTED",
    FEAR: "FEAR",
    HAPPY: "HAPPY",
    SAD: "SAD",
    SURPRISED: "SURPRISED",
    UNKNOWN: "UNKNOWN",
};
const LandmarkType = {
    chinBottom: "chinBottom",
    eyeLeft: "eyeLeft",
    eyeRight: "eyeRight",
    leftEyeBrowLeft: "leftEyeBrowLeft",
    leftEyeBrowRight: "leftEyeBrowRight",
    leftEyeBrowUp: "leftEyeBrowUp",
    leftEyeDown: "leftEyeDown",
    leftEyeLeft: "leftEyeLeft",
    leftEyeRight: "leftEyeRight",
    leftEyeUp: "leftEyeUp",
    leftPupil: "leftPupil",
    midJawlineLeft: "midJawlineLeft",
    midJawlineRight: "midJawlineRight",
    mouthDown: "mouthDown",
    mouthLeft: "mouthLeft",
    mouthRight: "mouthRight",
    mouthUp: "mouthUp",
    nose: "nose",
    noseLeft: "noseLeft",
    noseRight: "noseRight",
    rightEyeBrowLeft: "rightEyeBrowLeft",
    rightEyeBrowRight: "rightEyeBrowRight",
    rightEyeBrowUp: "rightEyeBrowUp",
    rightEyeDown: "rightEyeDown",
    rightEyeLeft: "rightEyeLeft",
    rightEyeRight: "rightEyeRight",
    rightEyeUp: "rightEyeUp",
    rightPupil: "rightPupil",
    upperJawlineLeft: "upperJawlineLeft",
    upperJawlineRight: "upperJawlineRight",
};
const KnownGenderType = {
    Female: "Female",
    Male: "Male",
    Nonbinary: "Nonbinary",
    Unlisted: "Unlisted",
};
const GenderType = {
    Female: "Female",
    Male: "Male",
};
const CelebrityRecognitionSortBy = {
    ID: "ID",
    TIMESTAMP: "TIMESTAMP",
};
const ChallengeType = {
    FACE_MOVEMENT_AND_LIGHT_CHALLENGE: "FaceMovementAndLightChallenge",
    FACE_MOVEMENT_CHALLENGE: "FaceMovementChallenge",
};
const QualityFilter = {
    AUTO: "AUTO",
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
    NONE: "NONE",
};
const OrientationCorrection = {
    ROTATE_0: "ROTATE_0",
    ROTATE_180: "ROTATE_180",
    ROTATE_270: "ROTATE_270",
    ROTATE_90: "ROTATE_90",
};
const ContentClassifier = {
    FREE_OF_ADULT_CONTENT: "FreeOfAdultContent",
    FREE_OF_PERSONALLY_IDENTIFIABLE_INFORMATION: "FreeOfPersonallyIdentifiableInformation",
};
const ContentModerationAggregateBy = {
    SEGMENTS: "SEGMENTS",
    TIMESTAMPS: "TIMESTAMPS",
};
const ContentModerationSortBy = {
    NAME: "NAME",
    TIMESTAMP: "TIMESTAMP",
};
const DatasetType = {
    TEST: "TEST",
    TRAIN: "TRAIN",
};
const ProjectAutoUpdate = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const CustomizationFeature = {
    CONTENT_MODERATION: "CONTENT_MODERATION",
    CUSTOM_LABELS: "CUSTOM_LABELS",
};
const DatasetStatus = {
    CREATE_COMPLETE: "CREATE_COMPLETE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
    UPDATE_COMPLETE: "UPDATE_COMPLETE",
    UPDATE_FAILED: "UPDATE_FAILED",
    UPDATE_IN_PROGRESS: "UPDATE_IN_PROGRESS",
};
const DatasetStatusMessageCode = {
    CLIENT_ERROR: "CLIENT_ERROR",
    SERVICE_ERROR: "SERVICE_ERROR",
    SUCCESS: "SUCCESS",
};
const UnsuccessfulFaceDeletionReason = {
    ASSOCIATED_TO_AN_EXISTING_USER: "ASSOCIATED_TO_AN_EXISTING_USER",
    FACE_NOT_FOUND: "FACE_NOT_FOUND",
};
const ProjectStatus = {
    CREATED: "CREATED",
    CREATING: "CREATING",
    DELETING: "DELETING",
};
const ProjectVersionStatus = {
    COPYING_COMPLETED: "COPYING_COMPLETED",
    COPYING_FAILED: "COPYING_FAILED",
    COPYING_IN_PROGRESS: "COPYING_IN_PROGRESS",
    DELETING: "DELETING",
    DEPRECATED: "DEPRECATED",
    EXPIRED: "EXPIRED",
    FAILED: "FAILED",
    RUNNING: "RUNNING",
    STARTING: "STARTING",
    STOPPED: "STOPPED",
    STOPPING: "STOPPING",
    TRAINING_COMPLETED: "TRAINING_COMPLETED",
    TRAINING_FAILED: "TRAINING_FAILED",
    TRAINING_IN_PROGRESS: "TRAINING_IN_PROGRESS",
};
const StreamProcessorStatus = {
    FAILED: "FAILED",
    RUNNING: "RUNNING",
    STARTING: "STARTING",
    STOPPED: "STOPPED",
    STOPPING: "STOPPING",
    UPDATING: "UPDATING",
};
const DetectLabelsFeatureName = {
    GENERAL_LABELS: "GENERAL_LABELS",
    IMAGE_PROPERTIES: "IMAGE_PROPERTIES",
};
const TextTypes = {
    LINE: "LINE",
    WORD: "WORD",
};
const UnsuccessfulFaceDisassociationReason = {
    ASSOCIATED_TO_A_DIFFERENT_USER: "ASSOCIATED_TO_A_DIFFERENT_USER",
    FACE_NOT_FOUND: "FACE_NOT_FOUND",
};
const FaceAttributes = {
    ALL: "ALL",
    DEFAULT: "DEFAULT",
};
const FaceSearchSortBy = {
    INDEX: "INDEX",
    TIMESTAMP: "TIMESTAMP",
};
const VideoJobStatus = {
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    SUCCEEDED: "SUCCEEDED",
};
const VideoColorRange = {
    FULL: "FULL",
    LIMITED: "LIMITED",
};
const LivenessSessionStatus = {
    CREATED: "CREATED",
    EXPIRED: "EXPIRED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    SUCCEEDED: "SUCCEEDED",
};
const LabelDetectionAggregateBy = {
    SEGMENTS: "SEGMENTS",
    TIMESTAMPS: "TIMESTAMPS",
};
const LabelDetectionSortBy = {
    NAME: "NAME",
    TIMESTAMP: "TIMESTAMP",
};
const MediaAnalysisJobFailureCode = {
    ACCESS_DENIED: "ACCESS_DENIED",
    INTERNAL_ERROR: "INTERNAL_ERROR",
    INVALID_KMS_KEY: "INVALID_KMS_KEY",
    INVALID_MANIFEST: "INVALID_MANIFEST",
    INVALID_OUTPUT_CONFIG: "INVALID_OUTPUT_CONFIG",
    INVALID_S3_OBJECT: "INVALID_S3_OBJECT",
    RESOURCE_NOT_FOUND: "RESOURCE_NOT_FOUND",
    RESOURCE_NOT_READY: "RESOURCE_NOT_READY",
    THROTTLED: "THROTTLED",
};
const MediaAnalysisJobStatus = {
    CREATED: "CREATED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    QUEUED: "QUEUED",
    SUCCEEDED: "SUCCEEDED",
};
const PersonTrackingSortBy = {
    INDEX: "INDEX",
    TIMESTAMP: "TIMESTAMP",
};
const TechnicalCueType = {
    BLACK_FRAMES: "BlackFrames",
    COLOR_BARS: "ColorBars",
    CONTENT: "Content",
    END_CREDITS: "EndCredits",
    OPENING_CREDITS: "OpeningCredits",
    SLATE: "Slate",
    STUDIO_LOGO: "StudioLogo",
};
const SegmentType = {
    SHOT: "SHOT",
    TECHNICAL_CUE: "TECHNICAL_CUE",
};
const Reason = {
    EXCEEDS_MAX_FACES: "EXCEEDS_MAX_FACES",
    EXTREME_POSE: "EXTREME_POSE",
    LOW_BRIGHTNESS: "LOW_BRIGHTNESS",
    LOW_CONFIDENCE: "LOW_CONFIDENCE",
    LOW_FACE_QUALITY: "LOW_FACE_QUALITY",
    LOW_SHARPNESS: "LOW_SHARPNESS",
    SMALL_BOUNDING_BOX: "SMALL_BOUNDING_BOX",
};
const LabelDetectionFeatureName = {
    GENERAL_LABELS: "GENERAL_LABELS",
};
const UnsearchedFaceReason = {
    EXCEEDS_MAX_FACES: "EXCEEDS_MAX_FACES",
    EXTREME_POSE: "EXTREME_POSE",
    FACE_NOT_LARGEST: "FACE_NOT_LARGEST",
    LOW_BRIGHTNESS: "LOW_BRIGHTNESS",
    LOW_CONFIDENCE: "LOW_CONFIDENCE",
    LOW_FACE_QUALITY: "LOW_FACE_QUALITY",
    LOW_SHARPNESS: "LOW_SHARPNESS",
    SMALL_BOUNDING_BOX: "SMALL_BOUNDING_BOX",
};
const StreamProcessorParameterToDelete = {
    ConnectedHomeMinConfidence: "ConnectedHomeMinConfidence",
    RegionsOfInterest: "RegionsOfInterest",
};

exports.$Command = client.Command;
exports.__Client = client.Client;
exports.RekognitionServiceException = RekognitionServiceException.RekognitionServiceException;
exports.AssociateFacesCommand = AssociateFacesCommand;
exports.Attribute = Attribute;
exports.BodyPart = BodyPart;
exports.CelebrityRecognitionSortBy = CelebrityRecognitionSortBy;
exports.ChallengeType = ChallengeType;
exports.CompareFacesCommand = CompareFacesCommand;
exports.ContentClassifier = ContentClassifier;
exports.ContentModerationAggregateBy = ContentModerationAggregateBy;
exports.ContentModerationSortBy = ContentModerationSortBy;
exports.CopyProjectVersionCommand = CopyProjectVersionCommand;
exports.CreateCollectionCommand = CreateCollectionCommand;
exports.CreateDatasetCommand = CreateDatasetCommand;
exports.CreateFaceLivenessSessionCommand = CreateFaceLivenessSessionCommand;
exports.CreateProjectCommand = CreateProjectCommand;
exports.CreateProjectVersionCommand = CreateProjectVersionCommand;
exports.CreateStreamProcessorCommand = CreateStreamProcessorCommand;
exports.CreateUserCommand = CreateUserCommand;
exports.CustomizationFeature = CustomizationFeature;
exports.DatasetStatus = DatasetStatus;
exports.DatasetStatusMessageCode = DatasetStatusMessageCode;
exports.DatasetType = DatasetType;
exports.DeleteCollectionCommand = DeleteCollectionCommand;
exports.DeleteDatasetCommand = DeleteDatasetCommand;
exports.DeleteFacesCommand = DeleteFacesCommand;
exports.DeleteProjectCommand = DeleteProjectCommand;
exports.DeleteProjectPolicyCommand = DeleteProjectPolicyCommand;
exports.DeleteProjectVersionCommand = DeleteProjectVersionCommand;
exports.DeleteStreamProcessorCommand = DeleteStreamProcessorCommand;
exports.DeleteUserCommand = DeleteUserCommand;
exports.DescribeCollectionCommand = DescribeCollectionCommand;
exports.DescribeDatasetCommand = DescribeDatasetCommand;
exports.DescribeProjectVersionsCommand = DescribeProjectVersionsCommand;
exports.DescribeProjectsCommand = DescribeProjectsCommand;
exports.DescribeStreamProcessorCommand = DescribeStreamProcessorCommand;
exports.DetectCustomLabelsCommand = DetectCustomLabelsCommand;
exports.DetectFacesCommand = DetectFacesCommand;
exports.DetectLabelsCommand = DetectLabelsCommand;
exports.DetectLabelsFeatureName = DetectLabelsFeatureName;
exports.DetectModerationLabelsCommand = DetectModerationLabelsCommand;
exports.DetectProtectiveEquipmentCommand = DetectProtectiveEquipmentCommand;
exports.DetectTextCommand = DetectTextCommand;
exports.DisassociateFacesCommand = DisassociateFacesCommand;
exports.DistributeDatasetEntriesCommand = DistributeDatasetEntriesCommand;
exports.EmotionName = EmotionName;
exports.FaceAttributes = FaceAttributes;
exports.FaceSearchSortBy = FaceSearchSortBy;
exports.GenderType = GenderType;
exports.GetCelebrityInfoCommand = GetCelebrityInfoCommand;
exports.GetCelebrityRecognitionCommand = GetCelebrityRecognitionCommand;
exports.GetContentModerationCommand = GetContentModerationCommand;
exports.GetFaceDetectionCommand = GetFaceDetectionCommand;
exports.GetFaceLivenessSessionResultsCommand = GetFaceLivenessSessionResultsCommand;
exports.GetFaceSearchCommand = GetFaceSearchCommand;
exports.GetLabelDetectionCommand = GetLabelDetectionCommand;
exports.GetMediaAnalysisJobCommand = GetMediaAnalysisJobCommand;
exports.GetPersonTrackingCommand = GetPersonTrackingCommand;
exports.GetSegmentDetectionCommand = GetSegmentDetectionCommand;
exports.GetTextDetectionCommand = GetTextDetectionCommand;
exports.IndexFacesCommand = IndexFacesCommand;
exports.KnownGenderType = KnownGenderType;
exports.LabelDetectionAggregateBy = LabelDetectionAggregateBy;
exports.LabelDetectionFeatureName = LabelDetectionFeatureName;
exports.LabelDetectionSortBy = LabelDetectionSortBy;
exports.LandmarkType = LandmarkType;
exports.ListCollectionsCommand = ListCollectionsCommand;
exports.ListDatasetEntriesCommand = ListDatasetEntriesCommand;
exports.ListDatasetLabelsCommand = ListDatasetLabelsCommand;
exports.ListFacesCommand = ListFacesCommand;
exports.ListMediaAnalysisJobsCommand = ListMediaAnalysisJobsCommand;
exports.ListProjectPoliciesCommand = ListProjectPoliciesCommand;
exports.ListStreamProcessorsCommand = ListStreamProcessorsCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListUsersCommand = ListUsersCommand;
exports.LivenessSessionStatus = LivenessSessionStatus;
exports.MediaAnalysisJobFailureCode = MediaAnalysisJobFailureCode;
exports.MediaAnalysisJobStatus = MediaAnalysisJobStatus;
exports.OrientationCorrection = OrientationCorrection;
exports.PersonTrackingSortBy = PersonTrackingSortBy;
exports.ProjectAutoUpdate = ProjectAutoUpdate;
exports.ProjectStatus = ProjectStatus;
exports.ProjectVersionStatus = ProjectVersionStatus;
exports.ProtectiveEquipmentType = ProtectiveEquipmentType;
exports.PutProjectPolicyCommand = PutProjectPolicyCommand;
exports.QualityFilter = QualityFilter;
exports.Reason = Reason;
exports.RecognizeCelebritiesCommand = RecognizeCelebritiesCommand;
exports.Rekognition = Rekognition;
exports.RekognitionClient = RekognitionClient;
exports.SearchFacesByImageCommand = SearchFacesByImageCommand;
exports.SearchFacesCommand = SearchFacesCommand;
exports.SearchUsersByImageCommand = SearchUsersByImageCommand;
exports.SearchUsersCommand = SearchUsersCommand;
exports.SegmentType = SegmentType;
exports.StartCelebrityRecognitionCommand = StartCelebrityRecognitionCommand;
exports.StartContentModerationCommand = StartContentModerationCommand;
exports.StartFaceDetectionCommand = StartFaceDetectionCommand;
exports.StartFaceSearchCommand = StartFaceSearchCommand;
exports.StartLabelDetectionCommand = StartLabelDetectionCommand;
exports.StartMediaAnalysisJobCommand = StartMediaAnalysisJobCommand;
exports.StartPersonTrackingCommand = StartPersonTrackingCommand;
exports.StartProjectVersionCommand = StartProjectVersionCommand;
exports.StartSegmentDetectionCommand = StartSegmentDetectionCommand;
exports.StartStreamProcessorCommand = StartStreamProcessorCommand;
exports.StartTextDetectionCommand = StartTextDetectionCommand;
exports.StopProjectVersionCommand = StopProjectVersionCommand;
exports.StopStreamProcessorCommand = StopStreamProcessorCommand;
exports.StreamProcessorParameterToDelete = StreamProcessorParameterToDelete;
exports.StreamProcessorStatus = StreamProcessorStatus;
exports.TagResourceCommand = TagResourceCommand;
exports.TechnicalCueType = TechnicalCueType;
exports.TextTypes = TextTypes;
exports.UnsearchedFaceReason = UnsearchedFaceReason;
exports.UnsuccessfulFaceAssociationReason = UnsuccessfulFaceAssociationReason;
exports.UnsuccessfulFaceDeletionReason = UnsuccessfulFaceDeletionReason;
exports.UnsuccessfulFaceDisassociationReason = UnsuccessfulFaceDisassociationReason;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateDatasetEntriesCommand = UpdateDatasetEntriesCommand;
exports.UpdateStreamProcessorCommand = UpdateStreamProcessorCommand;
exports.UserStatus = UserStatus;
exports.VideoColorRange = VideoColorRange;
exports.VideoJobStatus = VideoJobStatus;
exports.paginateDescribeProjectVersions = paginateDescribeProjectVersions;
exports.paginateDescribeProjects = paginateDescribeProjects;
exports.paginateGetCelebrityRecognition = paginateGetCelebrityRecognition;
exports.paginateGetContentModeration = paginateGetContentModeration;
exports.paginateGetFaceDetection = paginateGetFaceDetection;
exports.paginateGetFaceSearch = paginateGetFaceSearch;
exports.paginateGetLabelDetection = paginateGetLabelDetection;
exports.paginateGetPersonTracking = paginateGetPersonTracking;
exports.paginateGetSegmentDetection = paginateGetSegmentDetection;
exports.paginateGetTextDetection = paginateGetTextDetection;
exports.paginateListCollections = paginateListCollections;
exports.paginateListDatasetEntries = paginateListDatasetEntries;
exports.paginateListDatasetLabels = paginateListDatasetLabels;
exports.paginateListFaces = paginateListFaces;
exports.paginateListMediaAnalysisJobs = paginateListMediaAnalysisJobs;
exports.paginateListProjectPolicies = paginateListProjectPolicies;
exports.paginateListStreamProcessors = paginateListStreamProcessors;
exports.paginateListUsers = paginateListUsers;
exports.waitForProjectVersionRunning = waitForProjectVersionRunning;
exports.waitForProjectVersionTrainingCompleted = waitForProjectVersionTrainingCompleted;
exports.waitUntilProjectVersionRunning = waitUntilProjectVersionRunning;
exports.waitUntilProjectVersionTrainingCompleted = waitUntilProjectVersionTrainingCompleted;
Object.prototype.hasOwnProperty.call(schemas_0, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: schemas_0['__proto__']
    });

Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = schemas_0[k];
});
Object.prototype.hasOwnProperty.call(errors, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: errors['__proto__']
    });

Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = errors[k];
});
